pkg/subctl/cmd/join.go

```diff
@@ -95,6 +97,11 @@ func addJoinFlags(cmd *cobra.Command) {
 	cmd.Flags().BoolVar(&natTraversal, "natt", true, "enable NAT traversal for IPsec")
 	cmd.Flags().BoolVar(&disableNat, "disable-nat", false, "disable NAT for IPsec")
 	err := cmd.Flags().MarkDeprecated("disable-nat", "please use --natt=false instead")
+
+	cmd.Flags().BoolVar(&preferredServer, "preferred-server", false,
```

Comment:
In a simple two-cluster setup, if the user supplies `preferred-server` for both of them, how do we handle the scenario?

Reply:
yes, I'm coding that part :)

```
package libreswan

import (
	v1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
	"github.com/submariner-io/submariner/pkg/types"
	"k8s.io/klog"
)

type operationMode int

const (
	operationModeLegacy operationMode = iota
	operationModeServer
	operationModeClient
)

func (i *libreswan) calculateOperationMode(remoteEndpoint types.SubmarinerEndpoint) operationMode {
	defaultValue := false
	leftPreferred, err := i.localEndpoint.Spec.GetBackendBool(v1.PreferredServerConfig, &defaultValue)
	if err != nil {
		klog.Errorf("Error parsing local endpoint config: %s", err)
	}

	rightPreferred, err := remoteEndpoint.Spec.GetBackendBool(v1.PreferredServerConfig, nil)
	if err != nil {
		klog.Errorf("Error parsing remote endpoint config %q: %s", remoteEndpoint.Spec.CableName, err)
	}

	if rightPreferred == nil {
		return operationModeLegacy
	}

	if *leftPreferred && !*rightPreferred {
		return operationModeServer
	} else if *rightPreferred && !*leftPreferred {
		return operationModeClient
	}

	// At this point we didn't have enough info to decide, we break ties based on the cable name
	if i.localEndpoint.Spec.CableName > remoteEndpoint.Spec.CableName {
		return operationModeServer
	}

	return operationModeClient
}

```

Reply:
May be if none is preferred we can stick to the legacyMode... we could call it something else "bidirectional" or something